go-zen-chu/ubuntu-k8s-playbook
# Support k3s setup

## Why

- updating k8s with kubeadm is quite tough. By using k3s we can make k3s update more easy.

## What

- added k3s initializion

## QA, Evidence


setup successfully

```
(⎈ |kubernetes-admin@kubernetes:default)z> amasuda ~/ghq/github.com/go-zen-chu/ubuntu-k8s-playbook $ ansible-playbook -i hosts.yml k3s-control-plane.yml                                                                                                                               (git)-[create-by-k3s] 
SSH password: 
BECOME password[defaults to SSH password]: 

PLAY [k8s_control_planes] ************************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************************************************************************************************************************************************************************************
ok: [machine_control_plane1]

TASK [k3s-control-plane : Check k3s is installed] ************************************************************************************************************************************************************************************************************************************************************
ok: [machine_control_plane1]

TASK [k3s-control-plane : Install k3s if not installed] ******************************************************************************************************************************************************************************************************************************************************
skipping: [machine_control_plane1]

TASK [k3s-control-plane : Make sure k3s run as systemd] ******************************************************************************************************************************************************************************************************************************************************
ok: [machine_control_plane1]

TASK [k3s-control-plane : Fetch latest k3s generated token for k3s-agent] ************************************************************************************************************************************************************************************************************************************
ok: [machine_control_plane1]

TASK [k3s-control-plane : Fetch latest k3s generated token for k3s-agent] ************************************************************************************************************************************************************************************************************************************
ok: [machine_control_plane1]

PLAY RECAP ***************************************************************************************************************************************************************************************************************************************************************************************************
machine_control_plane1     : ok=5    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   

(⎈ |kubernetes-admin@kubernetes:default)z> amasuda ~/ghq/github.com/go-zen-chu/ubuntu-k8s-playbook $ ansible-playbook -i hosts.yml k3s-worker.yml                                                                                                                                      (git)-[create-by-k3s] 
SSH password: 
BECOME password[defaults to SSH password]: 

PLAY [k8s_workers] *******************************************************************************************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************************************************************************************************************************************************************************************
ok: [machine_worker2]
ok: [machine_worker1]

TASK [k3s-worker : Get k3s node token from local file] *******************************************************************************************************************************************************************************************************************************************************
ok: [machine_worker1]
ok: [machine_worker2]

TASK [k3s-worker : Check k3s is installed] *******************************************************************************************************************************************************************************************************************************************************************
ok: [machine_worker2]
ok: [machine_worker1]

TASK [k3s-worker : Install k3s if not installed] *************************************************************************************************************************************************************************************************************************************************************
skipping: [machine_worker1]
skipping: [machine_worker2]

TASK [k3s-worker : Make sure k3s run as systemd] *************************************************************************************************************************************************************************************************************************************************************
ok: [machine_worker2]
ok: [machine_worker1]

PLAY RECAP ***************************************************************************************************************************************************************************************************************************************************************************************************
machine_worker1            : ok=4    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   
machine_worker2            : ok=4    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0 

```
